server/src/main/java/com/octopus/teamcity/opentelemetry/server/BuildOverviewExtensionController.java

```diff
@@ -0,0 +1,126 @@
+import jetbrains.buildServer.web.openapi.*;
+import jetbrains.buildServer.web.util.WebUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.util.Date;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+import static com.octopus.teamcity.opentelemetry.common.PluginConstants.*;
+
+public class BuildOverviewExtensionController extends BaseController
+{
+    private static final Logger LOG = Logger.getLogger(BuildOverviewExtensionController.class.getName());
+
+    private final SBuildServer sBuildServer;
+    private final ProjectManager projectManager;
+    private final PluginDescriptor pluginDescriptor;
+    private BuildStorageManager buildStorageManager;
+
+    public BuildOverviewExtensionController(
+            @NotNull PagePlaces pagePlaces,
+            @NotNull SBuildServer sBuildServer,
+            @NotNull ProjectManager projectManager,
+            @NotNull PluginDescriptor pluginDescriptor,
+            @NotNull WebControllerManager controllerManager,
+            @NotNull BuildStorageManager buildStorageManager)
+    {
+        this.sBuildServer = sBuildServer;
+        this.projectManager = projectManager;
+
+        this.pluginDescriptor = pluginDescriptor;
+        this.buildStorageManager = buildStorageManager;
+
+        String url = "/otel-trace-url.html";
+
+        final SimplePageExtension sakuraPageExtension = new SimplePageExtension(pagePlaces);
+        sakuraPageExtension.setPluginName(PLUGIN_NAME);
+        sakuraPageExtension.setPlaceId(new PlaceId("SAKURA_BUILD_OVERVIEW"));
+        sakuraPageExtension.setIncludeUrl(url);
+        sakuraPageExtension.register();
+
+        // we're not showing on the classic UI, as we want it to appear on the build summary
+        // but if we do that, it appears twice on the Sakura UI, and it's in an extremely ugly
+        // iframe that makes me cry
+
+        //todo: reach out to JetBrains and ask about it
+
+        // final SimplePageExtension classicPageExtension = new SimplePageExtension(pagePlaces);
+        // classicPageExtension.setPluginName(PLUGIN_NAME);
+        // classicPageExtension.setPlaceId(PlaceId.BUILD_RESULTS_FRAGMENT);
+        // classicPageExtension.setIncludeUrl(url);
+        // classicPageExtension.register();
+
+        controllerManager.registerController(url, this);
+    }
+
+    @Nullable
+    @Override
+    protected ModelAndView doHandle(@NotNull HttpServletRequest request, @NotNull HttpServletResponse response) throws Exception {
+        final ModelAndView mv = new ModelAndView(pluginDescriptor.getPluginResourcesPath("buildOverviewExtension.jsp"));
+        boolean isSakuraUI = WebUtil.sakuraUIOpened(request);
+
+        //todo: I think we need to do some auto-refreshes until we have data?
```

Comment:
Does this need to be done as part of this PR?

Reply:
In an ideal world, yes. But I'm going with the theory that it's better to merge what we've got than wait for perfect.
It works "okay" for now, which is better than nothing, but this would make it better.